Library/Homebrew/utils/analytics.sh

```diff
@@ -0,0 +1,71 @@
+setup-analytics() {
+  if [[ -z "$HOMEBREW_ANALYTICS" || -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    return
+  fi
```

Comment:
Thanks for pointing this out, @apjanke! If I'm understanding you correctly, the fix that doesn't involve going back to `if` is either (to stay with the above example):

```shell
[[ -n "$HOMEBREW_AUTO_UPDATE" ]] || return 0
[[ -z "$HOMEBREW_NO_AUTO_UPDATE" ]] || return 0
```

or inverting the condition in the tests:

```shell
[[ -z "$HOMEBREW_AUTO_UPDATE" ]] && return
[[ -n "$HOMEBREW_NO_AUTO_UPDATE" ]] && return
```

Is this correct?